src/components/UserSelector.tsx

```diff
@@ -1,35 +1,80 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+
+type Props = {
+  users: User[],
+  selectedUser: User | null,
+  onSetSelectedUser: (state: User | null) => void,
+  onSetSelectedPost: (state: Post | null) => void,
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  selectedUser,
+  onSetSelectedUser,
+  onSetSelectedPost,
+}) => {
+  const [isShowingUsers, setIsShowingUsers] = useState(false);
+
+  const handlerSelectUser = (e: React.MouseEvent, user: User) => {
+    e.preventDefault();
+
+    setIsShowingUsers(false);
+    onSetSelectedUser(user);
+    onSetSelectedPost(null);
+  };
 
-export const UserSelector: React.FC = () => {
   return (
     <div
       data-cy="UserSelector"
-      className="dropdown is-active"
+      className={classNames(
+        'dropdown', { 'is-active': isShowingUsers },
+      )}
+      onBlur={() => setTimeout(() => setIsShowingUsers(false), 100)}
     >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setIsShowingUsers(!isShowingUsers)}
         >
-          <span>Choose a user</span>
+          {selectedUser === null || selectedUser.name === ''
+            ? (<span>Choose a user</span>)
+            : (<span>{selectedUser.name}</span>)}
 
           <span className="icon is-small">
             <i className="fas fa-angle-down" aria-hidden="true" />
           </span>
         </button>
       </div>
 
-      <div className="dropdown-menu" id="dropdown-menu" role="menu">
-        <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">Leanne Graham</a>
-          <a href="#user-2" className="dropdown-item is-active">Ervin Howell</a>
-          <a href="#user-3" className="dropdown-item">Clementine Bauch</a>
-          <a href="#user-4" className="dropdown-item">Patricia Lebsack</a>
-          <a href="#user-5" className="dropdown-item">Chelsey Dietrich</a>
-        </div>
-      </div>
+      {isShowingUsers
+        && (
+          <div className="dropdown-menu" id="dropdown-menu" role="menu">
+            <div className="dropdown-content">
+              {users.map(user => (
+                <a
+                  key={user.id}
+                  href={`#user-${user.id}`}
+                  className={classNames(
+                    'dropdown-item',
+                    {
+                      'is-active': selectedUser !== null
```

Comment:
Instead of repeating 'selectedUser !== null' check, you could include this check in the parent component or use optional chaining.